### PR TITLE
Sanitize symbols imported from R packages

### LINKED ISSUE #273: Cannot use functions with `.` (dot) in their names loaded using @rlibrary
MWE:
```
julia> @rlibrary PerformanceAnalytics
┌ Warning: RCall.jl:
│ Package PerformanceAnalytics (1.5.2) loaded.
│ Copyright (c) 2004-2018 Peter Carl and Brian G. Peterson, GPL-2 | GPL-3
│ https://github.com/braverock/PerformanceAnalytics
│ 
└ @ RCall ~/.julia/packages/RCall/Q4n8R/src/io.jl:110

julia> chart
chart.ACF                       chart.RollingCorrelation
chart.ACFplus                   chart.RollingMean
chart.Bar                       chart.RollingPerformance
chart.BarVaR                    chart.RollingQuantileRegression
chart.Boxplot                   chart.RollingRegression
chart.CaptureRatios             chart.Scatter
chart.Correlation               chart.SnailTrail
chart.CumReturns                chart.StackedBar
chart.Drawdown                  chart.TimeSeries
chart.ECDF                      chart.VaRSensitivity
chart.Events                    charts.Bar
chart.Histogram                 charts.BarVaR
chart.QQPlot                    charts.PerformanceSummary
chart.Regression                charts.RollingPerformance
chart.RelativePerformance       charts.RollingRegression
chart.RiskReturnScatter         charts.TimeSeries

julia> chart.Correlation
ERROR: UndefVarError: chart not defined
```

If there's no issue stopping it, I'd like to make a PR to sanitize the  symbols before being imported. Any thoughts?

### PR DESCRIPTION
**Pull request by jayvn**
Fixes #273 .

### CODE REVIEW

File: src/namespaces.jl [hunk 1]
```diff
@@ -11,23 +11,32 @@ cached_namespaces = Dict{String, Module}()
 
 """
 Import an R package as a julia module.
+Sanitizes the imported symbols by default, because otherwise certain symbols cannot be used.
+Eg: PerformanceAnalytics::charts.Bar in R becomes PerformanceAnalytics.charts_Bar in Julia
```
Comment (nalimilan) [inline src/namespaces.jl]: Needs backticks around code. Also an empty line after the short summary would be nice, and "Eg" should be "For example" or "E.g.".

Comment (jayvn) [reply src/namespaces.jl]: Changed and pushed. Thanks for the review.

**Comment by nalimilan** [comment]
Sorry for being a pain, but I wonder whether `sanitize` is the best name here. For example CSV.jl uses `normalizenames` for the same operation. Maybe it would make sense to use the same term for consistency?

**Comment by jayvn** [comment]
> Sorry for being a pain, but I wonder whether `sanitize` is the best name here. For example CSV.jl uses `normalizenames` for the same operation. Maybe it would make sense to use the same term for consistency?
Makes sense to be consistent. But wouldn't it be better to use `sanitize` everywhere since it's shorter which in turn I assume is because it's already associated with cosmetic effects unlike `normalize` (which is why I guess they added `normalize`+`names` for clarity)? 

Anyways, no problem changing the flag. I think I'll also have to change a few  `rcopy` definitions which use `sanitize` already too.

**Comment by randy3k** [comment]
I am open to rename `sanitize` to anything else. R's `read.csv`uses `check.names` fyi.

**Comment by jayvn** [comment]
Hi, 
Just checking up on this PR. Should I change `sanitize` to `normalizenames` or `check.names` ?

**Comment by nalimilan** [comment]
I'd be inclined to go with `normalizenames` as that's what CSV.jl uses.

**Comment by randy3k** [comment]
`check.names` is definitely not a good argument name in julia btw.

**Comment by jayvn** [comment]
OK. Should I change `sanitize` argument in the `rcopy` definitions as well? This'll break the existing API.

**Comment by randy3k** [comment]
Yes, we should change it with a deprecation notice.

**Comment by jayvn** [comment]
Sorry for the delay, having a bit of trouble with deprecating the function.

**Comment by jayvn** [comment]
I cannot `@deprecate` just for a keyword argument actually. I can write a redirecting function that issues a warning if there isn't a better way.

**Comment by simonbyrne** [comment]
One way to deprecate keyword args is to do something like
```julia
function rcopy(X; normalizenames::Bool=false, sanitize=nothing)
    if sanitize !== nothing
        Base.depwarn("The `sanitize` keyword argument is deprecated; use `normalizenames` instead", :rcopy)
        normalizenames = sanitize
    end
    ...
end
```

**Comment by jayvn** [comment]
I just pushed the exact same snippet @simonbyrne (except that I use `!=` instead of `!==`). @fredrikekre had suggested it to me on slack yesterday. Thanks anyway.

File: src/convert/dataframe.jl [hunk 2]
```diff
@@ -1,9 +1,15 @@
 # conversion methods for DataFrames
 
-function rcopy(::Type{T}, s::Ptr{VecSxp}; sanitize::Bool=true) where T<:AbstractDataFrame
+function rcopy(::Type{T}, s::Ptr{VecSxp};
+               normalizenames::Bool=true, 
+               sanitize=nothing) where T<:AbstractDataFrame
+    if sanitize != nothing
```
Comment (simonbyrne) [inline src/convert/dataframe.jl]: Best to use `!==` (the union splitting optimisation doesn't work quite as well with `!=`)

Comment (jayvn) [reply src/convert/dataframe.jl]: Changed and pushed.

**Comment by jayvn** [comment]
Hi,
Would like to see if there's any remaining issue?

File: src/convert/dataframe.jl [hunk 3]
```diff
@@ -1,9 +1,15 @@
 # conversion methods for DataFrames
 
-function rcopy(::Type{T}, s::Ptr{VecSxp}; sanitize::Bool=true) where T<:AbstractDataFrame
+function rcopy(::Type{T}, s::Ptr{VecSxp};
+               normalizenames::Bool=true, 
+               sanitize=nothing) where T<:AbstractDataFrame
```
Comment (nalimilan) [inline src/convert/dataframe.jl]:
```suggestion
               sanitize::Union{Bool,Nothing}=nothing) where T<:AbstractDataFrame
```

File: src/namespaces.jl [hunk 4]
```diff
@@ -11,23 +11,33 @@ cached_namespaces = Dict{String, Module}()
...
 Import an R package as a julia module.
+Sanitizes the imported symbols by default, because otherwise certain symbols cannot be used.
+
+E.g.: `PerformanceAnalytics::charts.Bar` in R becomes `PerformanceAnalytics.charts_Bar` in Julia
 ```
 gg = rimport("ggplot2")
 ```
 """
-function rimport(pkg::String, s::Symbol=:__anonymous__)
+function rimport(pkg::String, s::Symbol=:__anonymous__; normalizenames=true)
```
Comment (nalimilan) [inline src/namespaces.jl]:
```suggestion
function rimport(pkg::String, s::Symbol=:__anonymous__; normalizenames::Bool=true)
```

**Comment by jayvn** [comment]
Thanks for the review @nalimilan

**Comment by jayvn** [comment]
Hello,
Just checking to see if this is merged. Are there anymore issues?

**Pull request merged**